ingest/src/main/java/com/microsoft/azure/kusto/ingest/ManagedStreamingIngestClient.java

```diff
@@ -112,15 +162,51 @@ public IngestionResult ingestFromStream(StreamSourceInfo streamSourceInfo, Inges
         streamSourceInfo.validate();
         ingestionProperties.validate();
 
-        if (streamSourceInfo.isLeaveOpen()) {
-            throw new UnsupportedOperationException("LeaveOpen can't be true in ManagedStreamingIngestClient");
+        UUID sourceId = streamSourceInfo.getSourceId();
+        if (sourceId == null) {
+            sourceId = UUID.randomUUID();
         }
-        streamSourceInfo.setLeaveOpen(true);
+
+        byte[] streamingBytes;
 
         try {
-            for (int i = 0; i < MAX_RETRY_CALLS; i++) {
+            streamingBytes = IOUtils.readNBytes(streamSourceInfo.getStream(), MAX_STREAMING_SIZE_BYTES + 1);
+        } catch (IOException e) {
+            throw new IngestionClientException("Failed to read from stream.", e);
+        }
+
+        // NOTE: ByteArrayInputStream's close method is a no-op, so we don't need to close it.
+        ByteArrayInputStream byteArrayStream = new ByteArrayInputStream(streamingBytes);
+
+        if (streamingBytes.length > MAX_STREAMING_SIZE_BYTES) {
+            log.info("Stream size is greater than max streaming size (%d bytes). Falling back to queued.");
+            StreamSourceInfo managedSourceInfo = new StreamSourceInfo(new SequenceInputStream(byteArrayStream, streamSourceInfo.getStream()),
```

Comment:
I get that `streamSourceInfo` is already used in this scope, but this `StreamSourceInfo` is only used for streaming, right? When I think of `ManagedStreaming`, I think it can be used for either queued or streaming.

Reply:
Sorry, it can indeed be used for either. I missed that!